### PR TITLE
reenabled the debugger hook for pytest sessions

### PR DESCRIPTION
**Pull request by hackaugusto**
The previous implementation was hooking on the print_exception instead
of handle_error and allowed nested calls to pdb, which made it
impossible to quit the pdb session once the post_mortem was done,
because of that it was removed.

This tries to improve on the previous strategy, using the exposed
`handle_error` interface from the hub, which is intedend to be
overwritten, and by disallowing nested calls to pdb, which should both:

- allow a session to quit
- stop "randomly" jumping around different greenlets, if there are
multiple exceptions being raised

adding this utility back because it's really helpful for debugging

### CODE REVIEW

File: raiden/tests/conftest.py [hunk 1]
```diff
@@ -67,6 +67,45 @@ def pytest_addoption(parser):
...
+        def debugger(context, type, value, tb):
+            # Always print the exception, because once the pdb REPL is started
+            # we cannot retrieve it with `sys.exc_info()`.
+            #
+            # Using gevent's hub print_exception because it properly handles
+            # corner cases.
+            hub.print_exception(context, type, value, tb)
+
+            if not enabled:
+                enabled[1] = 1
```
Comment (ulope) [inline raiden/tests/conftest.py]: Can't you use `nonlocal enabled` here?

Comment (hackaugusto) [reply raiden/tests/conftest.py]: My heart still loves python2.7
